src/main/java/com/atlauncher/gui/tabs/tools/SkinUpdaterToolPanel.java

```diff
@@ -24,57 +24,43 @@
 
 import org.mini2Dx.gettext.GetText;
 
-import com.atlauncher.Data;
 import com.atlauncher.builders.HTMLBuilder;
-import com.atlauncher.evnt.listener.AccountListener;
-import com.atlauncher.evnt.manager.AccountManager;
 import com.atlauncher.gui.dialogs.ProgressDialog;
 import com.atlauncher.managers.DialogManager;
 import com.atlauncher.network.Analytics;
 
 @SuppressWarnings("serial")
-public class SkinUpdaterToolPanel extends AbstractToolPanel implements ActionListener, AccountListener {
+public class SkinUpdaterToolPanel extends AbstractToolPanel implements ActionListener {
 
-    public SkinUpdaterToolPanel() {
-        super(GetText.tr("Skin Updater"));
+    private final IToolsViewModel viewModel;
 
+    public SkinUpdaterToolPanel(IToolsViewModel viewModel) {
+        super(GetText.tr("Skin Updater"));
+        this.viewModel = viewModel;
         JLabel INFO_LABEL = new JLabel(new HTMLBuilder().center().split(70)
-                .text(GetText.tr("This tool will update all your accounts skins on the launcher.")).build());
+            .text(GetText.tr("This tool will update all your accounts skins on the launcher.")).build());
         MIDDLE_PANEL.add(INFO_LABEL);
         BOTTOM_PANEL.add(LAUNCH_BUTTON);
         LAUNCH_BUTTON.addActionListener(this);
-        AccountManager.addListener(this);
-        this.checkLaunchButtonEnabled();
-    }
-
-    private void checkLaunchButtonEnabled() {
-        LAUNCH_BUTTON.setEnabled(Data.ACCOUNTS.size() != 0);
+        viewModel.onSkinUpdaterEnabledChanged(LAUNCH_BUTTON::setEnabled);
     }
 
     @Override
     public void actionPerformed(ActionEvent e) {
         Analytics.sendEvent("SkinUpdater", "Run", "Tool");
```

Comment:
It seems off to me that half of these analytics events are sent within the tool panel itself and the rest in the view model, so there's no 1 clear place to know where to see the analytics events. I think no matter where they are, they should be consistently in one or the other

Reply:
Analytic events should only occur in view model.

The one purpose of the UI layer is to handle UI and updating the UI. Everything else should be delegated to the view model.

Moved calls in [9dcaa4f](https://github.com/ATLauncher/ATLauncher/pull/612/commits/9dcaa4f2c92f5d03a49b339968c205688ac731f2)